nifi-mock/src/main/java/org/apache/nifi/util/MockProcessorLog.java

```diff
@@ -16,20 +16,57 @@
  */
 package org.apache.nifi.util;
 
+import java.util.List;
+
 import org.apache.nifi.logging.ProcessorLog;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class MockProcessorLog implements ProcessorLog {
 
-    private final Logger logger;
+    private final CapturingLogger logger;
     private final Object component;
 
     public MockProcessorLog(final String componentId, final Object component) {
-        this.logger = LoggerFactory.getLogger(component.getClass());
+        this.logger = new CapturingLogger(LoggerFactory.getLogger(component.getClass()));
```

Comment:
Also, there may be a simpler way to do what I believe you are trying. Just look at here https://github.com/apache/nifi/pull/167/files#diff-d4270631731d05831ae336a5e6e50ad2R54. NOTE: that wit this approach you'll get access to the entire message after its being formatted, thus validating values in {..}